src/qml/qgismobileapp.qml

```diff
@@ -147,59 +147,90 @@ ApplicationWindow {
+        if (!active) {
+          var screenLocation = centroid.position;
+          var screenFraction = settings.value( "/QField/Digitizing/FreehandRecenterScreenFraction", 5 );
+          var threshold = Math.min( mainWindow.width, mainWindow.height ) / screenFraction;
+          if ( screenLocation.x < threshold || screenLocation.x > mainWindow.width - threshold ||
+               screenLocation.y < threshold || screenLocation.y > mainWindow.height - threshold )
+          {
+            mapCanvas.mapSettings.setCenter(mapCanvas.mapSettings.screenToCoordinate(screenLocation));
+          }
         }
+      }
 
-        onActiveChanged: {
-            if ( !active )
-              coordinateLocator.sourceLocation = undefined
-
+      onCentroidChanged: {
+        if (active) {
+          if (geometryEditorsToolbar.canvasClicked(centroid.position)) {
+            // needed to handle freehand digitizing of rings
+          } else {
+            currentRubberband.model.addVertex()
+          }
         }
+      }
+    }
 
-        onHoveredChanged: {
-            if ( !hovered )
-              coordinateLocator.sourceLocation = undefined
-        }
+    HoverHandler {
+      id: hoverHandler
+      enabled: !qfieldSettings.mouseAsTouchScreen && !parent.isBeingTouched
+      acceptedDevices: PointerDevice.Stylus | PointerDevice.Mouse
+      grabPermissions: PointerHandler.TakeOverForbidden
+
+      onPointChanged: {
+        // after a click, it seems that the position is sent once at 0,0 => weird
+        if (point.position !== Qt.point(0, 0))
+          coordinateLocator.sourceLocation = point.position
+      }
+
+      onActiveChanged: {
+        if ( !active )
+          coordinateLocator.sourceLocation = undefined
+
+      }
+
+      onHoveredChanged: {
+        if ( !hovered )
+          coordinateLocator.sourceLocation = undefined
+      }
     }
 
 
     Timer {
-        id: resetIsBeingTouchedTimer
-        interval: 750
-        repeat: false
+      id: resetIsBeingTouchedTimer
+      interval: 750
+      repeat: false
 
-        onTriggered: {
-            parent.isBeingTouched = false
-        }
+      onTriggered: {
+        parent.isBeingTouched = false
+      }
     }
     /* The second hover handler is a workaround what appears to be an issue with
      * Qt whereas synthesized mouse event would trigger the first HoverHandler even though
      * PointerDevice.TouchScreen was explicitly taken out of the accepted devices.
      */
     HoverHandler {
-        id: dummyHoverHandler
-        enabled: !qfieldSettings.mouseAsTouchScreen
-        acceptedDevices: PointerDevice.TouchScreen
-        grabPermissions: PointerHandler.TakeOverForbidden
-
-        onHoveredChanged: {
-            if ( hovered ) {
-                parent.isBeingTouched = true
-                resetIsBeingTouchedTimer.stop()
-            }
-            else {
-                resetIsBeingTouchedTimer.restart()
-            }
+      id: dummyHoverHandler
```

Comment:
Can we leave indentation at 4? That's what Qt's qml formatter tool imposes on us (and I hope it will be stable enough at some point to use)

Reply:
Done.